library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealTextButtonWithIcon.kt

```diff
@@ -0,0 +1,79 @@
+package com.epmedu.animeal.foundation.button
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.epmedu.animeal.foundation.theme.AnimealTheme
+
+@Composable
+fun AnimealTextButtonWithIcon(
+    icon: Painter,
+    text: String,
+    modifier: Modifier = Modifier,
+    color: Color = Color.Unspecified,
+    contentDescription: String? = null,
+    onClick: () -> Unit,
+) = AnimealTextButtonWithIcon(
+    modifier = modifier,
+    icon = { Icon(painter = icon, contentDescription = contentDescription, tint = color) },
+    text = text,
+    color = color,
+    onClick = onClick,
+)
+
+@Composable
+fun AnimealTextButtonWithIcon(
+    icon: ImageVector,
+    text: String,
+    modifier: Modifier = Modifier,
+    color: Color = Color.Unspecified,
+    contentDescription: String? = null,
+    onClick: () -> Unit,
+) = AnimealTextButtonWithIcon(
+    modifier = modifier,
+    icon = { Icon(imageVector = icon, contentDescription = contentDescription, tint = color) },
+    text = text,
+    color = color,
+    onClick = onClick,
+)
+
+@Composable
+internal fun AnimealTextButtonWithIcon(
+    icon: @Composable () -> Unit,
+    text: String,
+    modifier: Modifier = Modifier,
+    color: Color = Color.Unspecified,
+    enabled: Boolean = true,
+    onClick: () -> Unit,
+) = Row(
```

Comment:
What do you think if we will use `TextButton` in this case?

Reply:
Replaced.